rllib/examples/metrics/custom_metrics_in_env_runners.py

```diff
@@ -0,0 +1,339 @@
+        if (
+            self._env_runner_indices is not None
+            and env_runner.worker_index not in self._env_runner_indices
+        ):
+            return
+
+        # Erase the start position record.
+        del self._episode_start_position[episode.id_]
+
+        # Get all pacman y/x-positions from the episode.
+        yx_positions = episode.get_temporary_timestep_data("pacman_yx_pos")
+        # h x w
+        heatmap = np.zeros((80, 100), dtype=np.int32)
+        for yx_pos in yx_positions:
+            if yx_pos != (-1, -1):
+                heatmap[yx_pos[0], yx_pos[1]] += 1
+
+        # Create the actual heatmap image.
+        # Normalize the heatmap to values between 0 and 1
+        norm = Normalize(vmin=heatmap.min(), vmax=heatmap.max())
+        # Use a colormap (e.g., 'hot') to map normalized values to RGB
+        colormap = plt.get_cmap("coolwarm")  # try "hot" and "viridis" as well?
+        # Returns a (64, 64, 4) array (RGBA).
+        heatmap_rgb = colormap(norm(heatmap))
+        # Convert RGBA to RGB by dropping the alpha channel and converting to uint8.
+        heatmap_rgb = (heatmap_rgb[:, :, :3] * 255).astype(np.uint8)
+        # Log the image.
+        metrics_logger.log_value(
+            "pacman_heatmap",
+            heatmap_rgb,
+            reduce=None,
+            window=10,  # Log 10 images at most per EnvRunner/training iteration.
+        )
+
+        # Get the max distance travelled for this episode.
+        dist_travelled = np.max(
+            episode.get_temporary_timestep_data("pacman_dist_travelled")
+        )
+
+        # Log the max. dist travelled in this episode (window=100).
+        metrics_logger.log_value(
+            "pacman_max_dist_travelled",
+            dist_travelled,
+            # For future reductions (e.g. over n different episodes and all the
+            # data coming from other env runners), reduce by max.
+            reduce="max",
+            # Always keep the last 100 values and max over this window.
+            # Note that this means that over time, if the values drop to lower
+            # numbers again, the reported `pacman_max_dist_travelled` might also
+            # decrease again (meaning `window=100` makes this not a "lifetime max").
+            window=100,
+        )
+
+        # Log the average dist travelled per episode (window=200).
+        metrics_logger.log_value(
+            "pacman_mean_dist_travelled",
+            dist_travelled,
+            reduce="mean",  # <- default
+            # Always keep the last 200 values and average over this window.
+            window=200,
+        )
+
+        # Log the number of lifes (as EMA-smoothed; no window).
+        metrics_logger.log_value(
+            "pacman_lifes",
+            episode.get_infos(-1)["lives"],
+            reduce="mean",  # <- default (must be "mean" for EMA smothing)
+            ema_coeff=0.01,  # <- default EMA coefficient (`window` must be None)
+        )
```

Comment:
Maybe add a comment in regard to `metrics_logger.reduce` when to use it and why not here.